### DISCUSSION TITLE
Setup Arb.bind<MyClass> to never generate null values for the fields

### DISCUSSION THREAD

**Discussion by MessiasLima** [kotest/kotest#4033] [Q&A] [answered]
I have a class mapper that I want to test, and I'm using `Arb.bind` to generate a random class to test. However, my mapper class has some rules for when a field is null and I wanted to test it in two different use-cases.

This is what I'm trying to do
```kotlin
data class MyClass(
   val id: String,
   val name: String?
)

data class MyClassViewData(
   val id: String,
   val name: String
)

class MyClassViewDataMapper {
    fun map(myClass: MyClass) : MyClassViewData {
        return MyClassViewData(
            id = myClass.id,
            name = myClass.name ?: "Unknown"
        )
    }
}

class MyClassViewDataMapperTest {
    // ...
    @Test 
    fun `should map view data object`() {
        val fixtureMyClass = Arb.bind<MyClass>().next()
        val actual = mapperUnderTesting.map(fixtureMyClass)
        // Do the assertions
    }

    @Test 
    fun `should map view data object when name is null`() {
        val fixtureMyClass = Arb.bind<MyClass>().next().copy(name = null)
        val actual = mapperUnderTesting.map(fixtureMyClass)
        // Do the assertions
    }
}
```

This approach is supposed to work, but when I need to test the non-null one, it keeps creating null values on the `name` field making this test flaky.

Is it possible to disable the `null` edge case?

## Additional information
My project is a huge project with thousands of test cases. We are currently using the "Appmattus Kotlin fixture" to generate these fixture classes in runtime. The fixture generation is encapsulated in a function. So my main goal here is to be able to replace the current implementation by the Kotest Property testing without having to rewrite all these test classes.

**Comment by Kantis** [comment]
See https://kotest.io/docs/proptest/reflective-arbs.html, which describes how to instruct `Arb.bind` to use a particular Arb for a property. You could use that technique here to specify the Arb used for `MyClass::name`

**Reply by MessiasLima** [reply]
So I would need to update every test case, isn't? In my actual project, I have a function like that:

```kotlin
   protected inline fun <reified FixtureType : Any> fixture(): FixtureType {
        return Arb.bind<FixtureType> {
            bind(UUID::class to Arb.uuid())
        }.next()
    }

// So inside the test, I would write something like this

val fixtureMyClass = fixture<MyClass>()
```

So as far as I understoo, there is no global configuration that sets to not generate null values regardless the type. Is it right?

**Reply by Kantis** [reply, marked as answer]
I believe you could use `GlobalArbResolver.register(arb: Arb<A>)` to register a global default arb. In your case: 

```kotlin
object KotestConfig : AbstractProjectConfig() {
  override fun beforeProject() {
    GlobalArbResolver.register<String?>(Arb.string()) // register non-nullable arb for nullable strings
  }
}
```

**Reply by whck6** [reply]
You saved my life.
